cerbero/commands/uninstall.py

```diff
@@ -69,20 +69,14 @@ def run(self, config, args):
             # As a fallback, in case the uninstall target does not exist,
             # or in case it misses some of the installed files, remove
             # them manually
-            installed_files = cookbook.recipe_installed_files(recipe_name)
-            for f in [os.path.join(config.install_dir, f) for f in installed_files]:
-                # os.path.exists returns False for broken symbolic links
-                if os.path.exists(f) or os.path.islink(f):
-                    os.remove(f)
-
+            cookbook.recipe_remove_installed_files(recipe_name)
             cookbook.clean_recipe_status(recipe_name)
 
             # Lastly, remove the build directory
             build_dir = os.path.join(config.sources, recipe.package_name)
             if os.path.isdir(build_dir):
                 shutil.rmtree(build_dir)
 
-        shell.remove_empty_dirs(config.install_dir)
```

Comment:
why did you remove this one ? it's prettier to remove empty dirs, or maybe there are side effects ?

Reply:
Yep, indeed it has some nasty side effects. For instance, it may remove the `aclocal` dir that cerbero creates (even though it's emtpy). This causes the `autoools` recipe to not be able to install anything within that dir because it doesn't exist. One would expect an installation process to create the dirs that it needs, but that doesn't seem to be the case.

So, in the end I resigned and decided not to remove empty directories because of these side effects. We'll end up having empty and unnecessary dirs, but the alternative is recipes not installing correctly, so...

Reply:
Effectively it is better to no touch them ;)